README.md

```diff
@@ -1,275 +1,284 @@
-
-### 3. Variance Reduction
-- **CUPED** (Controlled-experiment Using Pre-Experiment Data):
-  - Use historical outcome data to reduce variance, choose any granularity
-  - Support for several covariates
-- **CUPAC** (Control Using Predictors as Covariates):
-  - Use any scikit-learn compatible estimator to predict the outcome with pre-experiment data
-
-## Quick Start
-
-### Power Analysis Example
+## 📖 What is cluster-experiments?
 
-```python
-import numpy as np
-import pandas as pd
-from cluster_experiments import PowerAnalysis, NormalPowerAnalysis
+`cluster-experiments` provides a complete toolkit for designing, running, and analyzing experiments, with particular strength in handling **clustered randomization** and complex experimental designs. Originally developed to address challenges in **switchback experiments** and scenarios with **network effects** where standard randomization isn't feasible, it has evolved into a general-purpose experimentation framework supporting both simple A/B tests and other randomization designs.
 
-# Create sample data
-N = 1_000
-df = pd.DataFrame({
-    "target": np.random.normal(0, 1, size=N),
-    "date": pd.to_datetime(
-        np.random.randint(
-            pd.Timestamp("2024-01-01").value,
-            pd.Timestamp("2024-01-31").value,
-            size=N,
-        )
-    ),
-})
+### Why "cluster"?
 
-# Simulation-based power analysis with CUPED
-config = {
-    "analysis": "ols",
-    "perturbator": "constant",
-    "splitter": "non_clustered",
-    "n_simulations": 50,
-}
-pw = PowerAnalysis.from_dict(config)
-power = pw.power_analysis(df, average_effect=0.1)
-
-# Normal approximation (faster)
-npw = NormalPowerAnalysis.from_dict({
-    "analysis": "ols",
-    "splitter": "non_clustered",
-    "n_simulations": 5,
-    "time_col": "date",
-})
-power_normal = npw.power_analysis(df, average_effect=0.1)
-power_line_normal = npw.power_line(df, average_effects=[0.1, 0.2, 0.3])
+The name reflects the library's origins in handling **cluster-randomized experiments**, where randomization happens at a group level (e.g., stores, cities, time periods) rather than at the individual level. This is critical when:
 
+- **Spillover/Network Effects**: Treatment of one unit affects others (e.g., testing driver incentives in ride-sharing)
+- **Operational Constraints**: You can't randomize individuals (e.g., testing restaurant menu changes)
+- **Switchback Designs**: Treatment alternates over time periods within the same unit
 
-# MDE calculation
-mde = npw.mde(df, power=0.8)
+While the library is aimed at these scenarios, it's equally capable of handling standard A/B tests with individual-level randomization.
 
-# MDE line with length
-mde_timeline = npw.mde_time_line(
-    df,
-    powers=[0.8],
-    experiment_length=[7, 14, 21]
-)
+---
+
+## Key Features
 
-print(power, power_line_normal, power_normal, mde, mde_timeline)
+### **Experiment Design**
+- **Power Analysis & Sample Size Calculation**
```

Comment:
this is not rendering well, it's just highlighting, I think you need more line breaks